File: apps/myjobhunter/backend/app/schemas/company/research_source.py
"""ResearchSource schemas — Phase 1 stub."""
import uuid
from datetime import datetime

from pydantic import BaseModel


class ResearchSourceRead(BaseModel):
    id: uuid.UUID
    company_research_id: uuid.UUID
    url: str
    source_type: str
    fetched_at: datetime

    model_config = {"from_attributes": True}

File: apps/myjobhunter/frontend/src/types/research-source.ts
/**
 * TypeScript model for a ResearchSource as returned by the MJH backend.
 * Mirrors `ResearchSourceResponse` in
 * apps/myjobhunter/backend/app/schemas/company/research_source_response.py.
 */
export interface ResearchSource {
  id: string;
  company_research_id: string;
  url: string;
  title: string | null;
  snippet: string | null;
  source_type: string;
  fetched_at: string;
  created_at: string;
}
